extensions/stats/config.h

```diff
@@ -0,0 +1,386 @@
+// response_code="200",
+// response_flags="-",
+// source_app="svc01-0",
+// source_principal="unknown",
+// source_version="v2",
+// source_workload="svc01-0v2",
+// source_workload_namespace="service-graph01"
+// }
+
+std::vector<Mapping> istioStandardDimensionsMappings() {
+  return {
+      MAPPING("reporter", (outbound ? vSource : vDest)),
+      // --> Peer info source
+      MAPPING("source_workload", source.workload_name()),
+      MAPPING("source_workload_namespace", source.namespace_()),
+      MAPPING("source_principal", requestInfo.source_principal),
+      MAPPING("source_app", source_labels["app"]),
+      MAPPING("source_version", source_labels["version"]),
+
+      // --> Peer info destination
+      MAPPING("destination_workload", dest.workload_name()),
+      MAPPING("destination_workload_namespace", dest.namespace_()),
+      MAPPING("destination_principal", requestInfo.destination_principal),
+      MAPPING("destination_app", dest_labels["app"]),
+      MAPPING("destination_version", dest_labels["version"]),
+
+      // --> Service info
+      MAPPING("destination_service_host", requestInfo.destination_service_host),
+      MAPPING("destination_service_name", dest.workload_name()),
+      MAPPING("destination_service_namespace", dest.namespace_()),
+
+      MAPPING("request_protocol", requestInfo.request_protocol),
+      MAPPING("response_code", requestInfo.response_code),
+      MAPPING_SYM("response_flags", requestInfo.response_flag, vDash),
+
+      MAPPING("connection_security_policy",
+              (outbound ? unknown : (requestInfo.mTLS ? vMTLS : vNone)))};
+}
+
+// StatGen is dimensioned using standard Istio dimensions.
+// The standard dimensions are defined in istioStandardDimensionsMappings.
+class StatGen {
+ public:
+  explicit StatGen(std::string name, MetricType metric_type,
+                   ValueExtractorFn value_fn)
+      : name_(name),
+        value_fn_(value_fn),
+        mappings_(istioStandardDimensionsMappings()) {
+    metric_ =
+        std::make_shared<Metric>(metric_type, name, mappings_.metricTags());
+  };
+
+  StatGen() = delete;
+  inline StringView name() const { return name_; };
+
+  SimpleStat resolve(bool outbound, const common::NodeInfo& source,
+                     const common::NodeInfo& dest,
+                     const Common::RequestInfo& request_info) const {
+    auto vals = mappings_.eval(outbound, source, dest, request_info);
+    auto metric_id = metric_->resolveWithFields(vals);
+    logDebug(absl::StrCat(__FILE__, "::", __FUNCTION__, ":", __LINE__, ":",
+                          metric_id, "/", source.name(), dest.name(),
+                          request_info.mTLS));
+    return SimpleStat(metric_id, value_fn_);
+  };
+
+ private:
+  std::string name_;
+  ValueExtractorFn value_fn_;
+  Mappings mappings_;
+  // TODO convert shared pointer to a direct struct once Metric has a
+  // constructor.
+  MetricSharedPtr metric_;
+};
+
+class RequestsTotal : public StatGen {
```

Comment:
It seems heavy to subclass StatsGen for each stat. Why not simply make each stat an instance of StatGen and pass in a simple lambda as valueFn for initialization?